src/extensions/askUserForConsent.ts

```diff
@@ -0,0 +1,100 @@
+import * as _ from "lodash";
+import * as clc from "cli-color";
+import * as marked from "marked";
+import TerminalRenderer = require("marked-terminal");
+
+import { FirebaseError } from "../error";
+import { logPrefix } from "../extensions/extensionsHelper";
+import * as iam from "../gcp/iam";
+import { promptOnce, Question } from "../prompt";
+import * as utils from "../utils";
+
+marked.setOptions({
+  renderer: new TerminalRenderer(),
+});
+
+/**
+ * Returns a string that will be displayed in the prompt to user.
+ * @param {string} role
+ * @return {!Promise<string>}
+ */
+export async function formatDescription(extensionName: string, projectId: string, roles: string[]) {
+  const question = `${clc.bold(
+    extensionName
+  )} will be granted the following access to project ${clc.bold(projectId)}`;
+  const results: string[] = await Promise.all(
+    roles.map(async (role: string) => {
+      return await retrieveRoleInfo(role);
+    })
```

Comment:
It's a really subtle thing, but you can omit `async` and `await` here - it'll still map to a list of promises which can be Promise.all'd

Reply:
Good catch!